imap_processing/codice/codice_l1a_ialirt_hi.py

```diff
@@ -137,47 +146,34 @@ def l1a_ialirt_hi(unpacked_dataset: xr.Dataset, lut_file: Path) -> xr.Dataset:
         *collapse_shape,
     )
 
-    species_chunk_sizes = [
-        len(species_data[species]["min_energy"]) for species in species_names
-    ]
-
-    start_idx = 0
+    species_chunk_sizes = [len(species_data[first_species]["min_energy"])]
 
-    for index, (species_name, data) in enumerate(species_data.items()):
-        centers, _, _ = get_energy_info(data)
+    l1a_dataset = l1a_dataset.assign_coords(
+        {f"energy_{first_species}": (f"energy_{first_species}", np.array(centers))}
+    )
 
-        l1a_dataset = l1a_dataset.assign_coords(
-            {
-                f"energy_{species_name}": xr.DataArray(
-                    np.array(centers),
-                    dims=(f"energy_{species_name}",),
-                )
-            }
-        )
+    chunk_size = species_chunk_sizes[0]
+    end_idx = chunk_size * n_spins
 
-        chunk_size = species_chunk_sizes[index]
-        end_idx = start_idx + chunk_size * n_spins
+    species_array = decompressed_data[:, 0:end_idx]
```

Comment:
Do you need to make sure you are starting at the beginning of a sequence here? I think decompressed_data is for every ialirt packet isn't it, but CoDICE is combining a bunch of packets together to get their full 3456 byte buffer. I am wondering if this should somehow be something like a 1d array that has been concatenated together so that you have a long byte chunk here.